backend/twitterbot/tournesolbot.py

```diff
@@ -0,0 +1,152 @@
+            title=video_title,
+            twitter_account=twitter_account,
+            n_comparison=video.rating_n_ratings,
+            n_contributor=video.rating_n_contributors,
+            crit1=crit_dict[crit1[0]],
+            crit2=crit_dict[crit2[0]],
+            video_id=video_id,
+        )
+
+    return tweet_text
+
+
+def get_video_recommendations(language):
+    """Find a list of good video to recommend with some quality criteria."""
+
+    # Filter videos with some quality criteria
+    tweetable_videos = Entity.objects.filter(
+        add_time__lte=time_ago(days=settings.DAYS_TOO_RECENT),
+        metadata__publication_date__gte=time_ago(
+            days=settings.DAYS_TOO_OLD
+        ).isoformat(),
+        rating_n_contributors__gt=settings.MIN_NB_CONTRIBUTORS,
+        rating_n_ratings__gt=settings.MIN_NB_RATINGS,
+        metadata__language=language,
+        tournesol_score__gte=settings.MIN_TOURNESOL_SCORE,
+        criteria_scores__criteria="reliability",
+        criteria_scores__score__gt=settings.MIN_RELIABILITY_SCORE,
+        tweets=None,
+    )
+
+    # Exclude video recently tweeted by the same uploader
+    exclude_uploader = TweetInfo.objects.filter(
+        datetime_tweet__gte=time_ago(days=settings.DAYS_ALREADY_TWEETED_UPLOADER)
+    ).values_list("video__metadata__uploader", flat=True)
+
+    tweetable_videos = tweetable_videos.exclude(metadata__uploader__in=exclude_uploader)
+
+    return list(tweetable_videos)
+
+
+def select_a_video(tweetable_videos):
+    """Select a video to tweet."""
+
+    tournesol_score_list = [v.tournesol_score for v in tweetable_videos]
+
+    # Chose a random video weighted by tournesol score
+    selected_video = random.choices(tweetable_videos, weights=tournesol_score_list)[0]
+
+    return selected_video
+
+
+def tweet_video_recommendation(bot_name, debug=True):
+    """Tweet a video recommendation.
+
+    Args:
+        bot_name (str): The name of the bot.
+        debug (bool): If True, a confirmation will be asked before tweeting it.
+
+    """
+
+    twitterbot = TwitterBot(bot_name)
+    twitterbot.authenticate()
+
+    tweetable_videos = get_video_recommendations(language=twitterbot.language)
+    if not tweetable_videos:
+        print("No video reach the criteria to be tweeted today!!!")
+        return
+
+    video = select_a_video(tweetable_videos)
+    tweet_text = prepare_tweet(video)
+
+    if debug:
+
+        print("Today's video to tweet will be:")
+        print(tweet_text)
+
+        confirmation = input("\nWould you like to tweet that? (y/n)")
+
+        if confirmation == "y":
+
+            resp = twitterbot.api.update_status(tweet_text)
+
+            # Add the video to the TweetInfo table
+            TweetInfo.objects.create(
+                video=video,
+                tweet_id=resp.id,
+                datetime_tweet=twitterbot.api.now(),
+                bot_name=bot_name,
+            )
```

Comment:
PEP20, flat is better than nested.

```suggestion
    if not debug:
        return

    print("Today's video to tweet will be:")
    print(tweet_text)
    
    confirmation = input("\nWould you like to tweet that? (y/n)")
    if confirmation != "y":
        return

    resp = twitterbot.api.update_status(tweet_text)
    
    # Add the video to the TweetInfo table
    TweetInfo.objects.create(
        video=video,
        tweet_id=resp.id,
        datetime_tweet=twitterbot.api.now(),
        bot_name=bot_name,
    )
```

Reply:
Thanks, it is indeed better as you propose :)